### PR TITLE
[2241] Make the pipeline pass the API url as part of the message to SNS

### PR DESCRIPTION
**Pull request by cosa65**
# Description

# Details
#### URL to issue
https://biomage.atlassian.net/browse/BIOMAGE-2241

https://ui-martinfosco-pipeline68.scp-staging.biomage.net/
#### Link to staging deployment URL (or set N/A)
N/A

#### Integration test branch
master

# Merge checklist
Your changes will be ready for merging after all of the steps below have been completed.

### Code updates
Have best practices and ongoing refactors being observed in this PR
- [ ] Migrated any selector / reducer used to the new format.

### Manual/unit testing
- [ ] Tested changes using InfraMock locally **or** no tests required for change, e.g. Kubernetes chart updates.
- [ ] Validated that current unit tests for code work as expected and are sufficient for code coverage **or** no unit tests required for change, e.g. documentation update.
- [ ] Unit tests written **or** no unit tests required for change, e.g. documentation update.

### Integration testing
**You must check the box below** to run integration tests on the latest commit on your PR branch.
Integration tests have to pass before the PR can be merged. Without checking the box, your PR
**will not pass** the required status checks for merging.

- [ ] Started end-to-end tests on the latest commit.

### Documentation updates
- [ ] Relevant Github READMEs updated **or** no GitHub README updates required.
- [ ] Relevant Wiki pages created/updated **or** no Wiki updates required.

### Optional
- [ ] Staging environment is unstaged before merging.
- [ ] Photo of a cute animal attached to this PR.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -296,39 +296,36 @@ jobs:
...
           API_TOKEN_GITHUB: ${{ secrets.API_TOKEN_GITHUB }}
         with:
-          source_file: '${{ steps.fill-metadata.outputs.deployment-name }}.yaml'
-          destination_repo: '${{ github.repository_owner }}/releases'
-          destination_folder: '${{ steps.fill-metadata.outputs.kubernetes-env }}'
-          user_email: 'ci@biomage.net'
-          user_name: 'Biomage CI/CD'
+          source_file: "${{ steps.fill-metadata.outputs.deployment-name }}.yaml"
+          destination_repo: "${{ github.repository_owner }}/releases"
+          destination_folder: "${{ steps.fill-metadata.outputs.kubernetes-env }}"
```
Comment (ivababukova) [inline .github/workflows/ci.yaml:306]: you really need to resolve those single vs double quotes differences

File: pipeline-runner/R/init-functions.R [hunk 2]
```diff
@@ -77,22 +77,18 @@ load_config <- function(development_aws_server) {
...
-        config$api_url <- paste0("https://api-", sandbox, ".", domain_name)
-      }
-      if (config$cluster_env == "production") {
-        config$api_url <- paste0("https://api.", domain_name)
-      }
+  if (config$cluster_env == "staging") {
+    config$api_url <- paste0("https://api-", sandbox, ".", domain_name)
+  }
+  if (config$cluster_env == "production") {
+    config$api_url <- paste0("https://api.", domain_name)
```
Comment (ivababukova) [inline pipeline-runner/R/init-functions.R:91]: Seems like with this code we will no longer use the `".svc.cluster.local:3000"` domain name. Is this true?

Comment (cosa65) [reply pipeline-runner/R/init-functions.R:91]: After the standup feedback I am switching it back so we use the internal url when we don't use the lambda

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (cosa65) [reply .github/workflows/ci.yaml:306]: yes, the linter does it automatically although honestly I don't know if it is necessarily wrong to have linting active here, it seems like before this we had a mix of single quotes and no quotes, should I change it?

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -296,39 +296,36 @@ jobs:
...
         env:
           DEPLOYMENT_NAME: ${{ steps.fill-metadata.outputs.deployment-name }}
           REF_ID: ${{ needs.build-docker.outputs.ref-id }}
         run: |-
           mv $DEPLOYMENT_NAME.yaml $REF_ID.yaml
 
       - name: Push staging deployment template to releases
-        if:
-          (github.event_name == 'pull_request' || github.event_name == 'push') && matrix.environment == 'staging'
+        if: (github.event_name == 'pull_request' || github.event_name == 'push') && matrix.environment == 'staging'
```
Comment (kafkasl) [inline .github/workflows/ci.yaml:319]: why this change?

**Review by kafkasl — CHANGES_REQUESTED** [review]

**Review by ivababukova — COMMENTED** [review]
I don't understand how these changes pass in the API URL in the message to SNS -- let's talk about this over a huddle

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (cosa65) [reply .github/workflows/ci.yaml:306]: ok

Comment (cosa65) [reply .github/workflows/ci.yaml]: > which linter are you talking about?

this one:
image

Comment (cosa65) [reply .github/workflows/ci.yaml]: I set it to single quote now, let me know if it's fine

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (cosa65) [reply .github/workflows/ci.yaml:319]: probably more auto changes from vscode, maybe it's just easier for me to try to disable whatever kind of format on save I have right now and remove the changes in ci

**Comment by cosa65** [comment]
> I don't understand how these changes pass in the API URL in the message to SNS -- let's talk about this over a huddle

sure, in case it helps I'll write it here too:

I set `public_api_url` and then pick it up and pass it in the 3 calls to sns: (in `handle_data.R`).

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (kafkasl) [reply .github/workflows/ci.yaml]: ok, we should try to have all the same settings using a workspace instead of custom settings

**Review by ivababukova — APPROVED** [review]

**Review by kafkasl — APPROVED** [review]

**Pull request merged**